packages/rsocket-websocket-client/src/RSocketWebSocketClient.js

```diff
@@ -37,22 +36,29 @@ export type ClientOptions = {|
  * A WebSocket transport client for use in browser environments.
  */
 export default class RSocketWebSocketClient implements DuplexConnection {
+  _url: string;
   _encoders: ?Encoders<*>;
   _options: ClientOptions;
+  _wsCreator: (url: string, options: ClientOptions) => any;
   _receivers: Set<ISubscriber<Frame>>;
   _senders: Set<ISubscription>;
   _socket: ?WebSocket;
   _status: ConnectionStatus;
   _statusSubscribers: Set<ISubject<ConnectionStatus>>;
 
-  constructor(options: ClientOptions, encoders: ?Encoders<*>) {
-    this._encoders = encoders;
-    this._options = options;
-    this._receivers = new Set();
-    this._senders = new Set();
-    this._socket = null;
-    this._status = CONNECTION_STATUS.NOT_CONNECTED;
-    this._statusSubscribers = new Set();
+  constructor(url: string,
+    options: ClientOptions,
+    wsCreator: (url: string, options: ClientOptions) => any,
```

Comment:
Can we make this part of `ClientOptions` and make sure it's properly optional? The return type should also be `WebSocket`

Reply:
`wsCreator`'s return type is now `WebSocket`, and is an optional field of the `ClientOptions` type.

Previously the type of `wsCreator`'s second argument was `ClientOptions`, in order to pass e.g. headers to the websocket. But now that `wsCreator` is itself in `ClientOptions`, I've revised its type signature to be `wsCreator(url: string, wsOptions: {}) => WebSocket`. I wasn't sure how to type `wsOptions`, or if it's fine the way it now is. `wsOptions: {}` is now also an optional field in `ClientOptions`.